### PR TITLE
Fix: Provide missing required php version in composer

### PR DESCRIPTION
**Pull request by prisis**
- [x] runs composer validate in before_install section on Travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ cache:
 
 before_install:
   - mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || echo "xdebug not available"
+  - composer validate
   - travis_retry composer self-update
   - travis_retry composer config --unset platform.php
```
Comment (jwage) [inline .travis.yml]: I think we need to remove this line now, right?

Comment (prisis) [reply .travis.yml]: oh yes

**Comment by jwage** [comment]
This isn't related to your changes (I don't think) but can you run ./vendor/bin/phpstan a locally and fix the errors reported? This is what is erroring in the travis-ci build.

**Comment by prisis** [comment]
@jwage done

**Comment by jwage** [comment]
Thanks!

File: composer.json [hunk 2]
```diff
@@ -42,7 +43,6 @@
     "bin": [
         "bin/changelog-generator"
     ],
-    "platform": {
-        "php": "7.1.x"
```
Comment (Majkl578) [inline composer.json]: This thing has different purpose than `require.php` - it tells composer to install dependencies compatible with PHP 7.1, whatever PHP versions are allowed. It's not a replacement of `require.php`.

File: composer.json [hunk 3]
```diff
@@ -42,7 +43,6 @@
...
-    }
+    "minimum-stability": "dev",
```
Comment (Majkl578) [inline composer.json]: Not really a fan here. Whenever dev version is required, `@dev` override should be used on per-dependency level, not globally.

File: composer.json (hunk 2 above)
Comment (jwage) [reply composer.json]: This is what this is for, right? https://github.com/jwage/changelog-generator/blob/master/build-phar.sh#L6

Comment (Majkl578) [reply composer.json]: Yes, one of those use cases. :)

**Pull request merged**